crowbar_framework/app/models/logging_service.rb

```diff
@@ -62,18 +62,19 @@ def transition(inst, name, state)
     #
     if state == "discovered"
       @logger.debug("Logging transition: discovered state for #{name} for #{state}")
-      db = Proposal.where(barclamp: "logging", name: inst).first
       role = RoleObject.find_role_by_name "logging-config-#{inst}"
+      node = NodeObject.find_node_by_name name
 
-      if role.override_attributes["logging"]["elements"]["logging-server"].nil? or
-         role.override_attributes["logging"]["elements"]["logging-server"].empty?
-        @logger.debug("Logging transition: make sure that logging-server role is on first: #{name} for #{state}")
-        result = add_role_to_instance_and_node("logging", inst, name, db, role, "logging-server")
-      else
-        node = NodeObject.find_node_by_name name
-        unless node.role? "logging-server"
-          @logger.debug("Logging transition: make sure that logging-client role is on all nodes but first: #{name} for #{state}")
-          result = add_role_to_instance_and_node("logging", inst, name, db, role, "logging-client")
+      Proposal.where(barclamp: "logging", name: inst).first do |db|
+        if role.override_attributes["logging"]["elements"]["logging-server"].nil? or
+          role.override_attributes["logging"]["elements"]["logging-server"].empty?
```

Comment:
Style/MultilineOperationIndentation: Use 4 (not 2) spaces for indenting a condition in an `if` statement spanning multiple lines.